Docs: Relabel and move explanation and how-tos around OAuth (Diátaxis) (#9834)
readthedocs/readthedocs.org: docs/user/connected-accounts.rst

Connecting your Git repository
==============================

In this article, we explain how connecting your Read the Docs account to one of the supported Git providers
automatically configures your Git repository and your Read the Docs project.

✅️ Signed up with your Git provider?
  If you signed up or logged in to Read the Docs with your GitHub, Bitbucket, or GitLab
  credentials, you're all done. Your account is connected and the integration explained
  below is enabled.
⬇️ Signed up with your email address?
  If you have signed up to Read the Docs with your email address,
  you can add the connection to the Git provider afterwards.

>   Read more about how it works below.

Reviewer: We should link at least one of these `below`'s.

Author: This has been fixed by adding a direct link to the how-to in this step. I think that it makes sense - either it's the prerequisite for the explanation to be relevant or it's the "action item" of understanding the explanation and wanting to switch on the integration.

If you are going to import repositories from GitHub, Bitbucket, or GitLab,

> we recommend that you connect your Read the Docs account to your repository provider.

Reviewer 2: We are using different names to refer to the same thing in these documents.

* repository provider
* git provider
* repository host

I'd say we should use only one to refer to this because it may confuse the users.

Author: Thanks! Yes, this decision should really be made now that we have a Style Guide to consolidate it in: https://dev.readthedocs.io/en/latest/style-guide.html#word-list

We decided to say "Git repository" instead of VCS, but we didn't add anything that reasonably covers the "platform/host/provider" in the Style Guide.

I'll forward this in the open PR for the style guide.. adding a task for this PR to implement whatever we decide :+1:

Author: Added a suggestion for using "Git provider" to https://github.com/readthedocs/readthedocs.org/pull/9840 - let's decide there.

Connecting your account allows for:

* Easy import of your repositories
* Automatic configuration of your repository :doc:`/integrations`
  which allow Read the Docs to build your docs on every change to your repository

> * Logging into Read the Docs with your GitHub, Bitbucket, or GitLab credentials

Reviewer: These should all end with a period, and we should note that in the style guide :)

Author: Fixed :)

Quick eyes sometimes don't get old structures updated.

.. seealso::

   :doc:`/guides/connecting-git-account`
     Here are the steps to connect your Read the Docs account to your Git repository provider.

   :ref:`intro/import-guide:Manually import your docs`
     Using a different provider?
     Read the Docs still supports other providers such as Gitea or GitHub Enterprise.
     In fact, any Git repository URL can be configured manually.

.. tip::

>    A single Read the Docs account can connect to many different Git providers.
>    This means that you can gather all your documentation projects under the same roof.

Reviewer: ```suggestion
   A single Read the Docs account can connect to many different Git providers.
   This allows you to have a single login for all your various identifies. 
```

This is an odd phrasing..

For similar reasons to those above for GitHub, we request permissions for:

> * Administing your repositories (to add a read-only deployment key)

Reviewer: ```suggestion
* Administering your repositories (to add a read-only deployment key)
```
No idea what the original word here is :)

Author: Original word was "Administer", -- "Administing" sounds fun though :)

> * Administering your repositories (to add a read-only deployment key)

Author: ```suggestion
* Administering your repositories (to add webhooks)
```

I see it's similar to GitHub: https://developer.atlassian.com/cloud/bitbucket/bitbucket-cloud-rest-api-scopes/#repository-admin
